hack/api-docs/Makefile

```diff
@@ -45,13 +48,23 @@ image:
 build: image generate $(SOURCES)
 	mkdir -p $(GENROOT)
 	$(DOCKER) run \
-		--mount type=bind,source=$(SRCDIR),target=/docs \
-		--mount type=bind,source=$(CFGFILE),target=/mkdocs.yml \
-		--mount type=bind,source=$(GENROOT),target=/site \
+	    --mount type=bind,source=$(ROOT),target=/repo \
 		--sig-proxy=true \
 		--rm \
+		--user $(UID):$(GID) \
 		$(MKDOCS_IMAGE) \
-		/bin/bash -c "cd / && $(MKDOCS) build; find /site -exec chown $(UID):$(GID) {} \;"
+		/bin/bash -c "cd /repo && git config user.name "Docs" && git config user.email "docs@external-secrets.io" && $(MIKE) deploy --update-aliases -F hack/api-docs/mkdocs.yml $(DOCS_VERSION) $(DOCS_ALIAS);"
+
+.PHONY: build.publish
+build.publish: image generate $(SOURCES)
+	mkdir -p $(GENROOT)
+	$(DOCKER) run \
+	    --mount type=bind,source=$(ROOT),target=/repo \
+		--sig-proxy=true \
+		--rm \
+		--user $(UID):$(GID) \
+		$(MKDOCS_IMAGE) \
+		/bin/bash -c "cd /repo && git config user.name "Docs" && git config user.email "docs@external-secrets.io" && $(MIKE) deploy --update-aliases -p -F hack/api-docs/mkdocs.yml $(DOCS_VERSION) $(DOCS_ALIAS);"
```

Comment:
We can move the `git config --global user.(name|email)` to the Dockerfile to keep it a bit simpler

Reply:
Will do! I think I understood what was missing for the `not seeing the latest` banner. I'm going to test it, and in the following I'll move the git config.

Reply:
it seems my latest commit is not being taken into account by github... but changes are [here](https://github.com/external-secrets/external-secrets/pull/663/commits/ade03c6a03f44c4a2db09ad8326c0ded20736a84)

Reply:
The publish target still has the `git config` things

Reply:
forgot about this one :s